src/a11y-dialog.ts

```diff
@@ -3,22 +3,27 @@ import focusableSelectors from 'focusable-selectors'
 const TAB_KEY = 'Tab'
 const ESCAPE_KEY = 'Escape'
 
-/**
- * @typedef { import ('./a11y-dialog').A11yDialog } A11yDialogType
- * @typedef { import('./a11y-dialog').EventType } EventType
- * @typedef { import('./a11y-dialog').EventHandler } EventHandler
- * @typedef { import('./a11y-dialog').ListenersRecord } ListenersRecord
- */
+export type A11yDialogEvent = 'show' | 'hide' | 'destroy' | 'create'
+export type A11yDialogInstance = InstanceType<typeof A11yDialog>
+export type A11yDialogEventHandler = (node: Element, event?: Event) => void
+
+type ListenersRecord = Partial<
+  Record<A11yDialogEvent, A11yDialogEventHandler[]>
+>
 
-/** @type A11yDialogType */
 export default class A11yDialog {
-  /** @param {element} HTMLElement */
-  constructor(element) {
+  private $el: HTMLElement
+  private id: string
+  private openers: HTMLElement[]
+  private closers: HTMLElement[]
+  private listeners: ListenersRecord = {}
+  private previouslyFocused: HTMLElement | null = null
+
+  public shown = false
+
+  constructor(element: HTMLElement) {
     this.$el = element
     this.id = this.$el.getAttribute('data-a11y-dialog') || this.$el.id
-    this.listeners = {}
-    this.previouslyFocused = null
-    this.shown = false
```

Comment:
Since TypeScript's strict mode requires us to declare all the things we assign in the constructor, I moved these assignments to lines 19 - 22.